tests/server/wildfly/src/test/java/org/hawkular/apm/tests/wildfly/zipkin/SpanAnalyticsITest.java

```diff
@@ -0,0 +1,253 @@
+                        SpanServiceITest.millisToMicroSeconds(1), SpanServiceITest.millisToMicroSeconds(30)));
+        rootSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(1L));
+        rootSpan.setDuration(SpanServiceITest.calculateDuration(1L, 30L));
+        rootSpan.setTraceId("root");
+        rootSpan.setId("root");
+        rootSpan.setName("get");
+
+        BinaryAnnotation urlServiceABinAnn = new BinaryAnnotation();
+        urlServiceABinAnn.setKey(Constants.ZIPKIN_BIN_ANNOTATION_HTTP_URL);
+        urlServiceABinAnn.setValue("http://localhost/serviceA");
+
+        Span clientSpan = new Span(Arrays.asList(urlServiceABinAnn),
+                SpanServiceITest.clientAnnotation(SpanServiceITest.millisToMicroSeconds(5),
+                        SpanServiceITest.millisToMicroSeconds(15)));
+        clientSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(5));
+        clientSpan.setDuration(SpanServiceITest.calculateDuration(5L, 15L));
+        clientSpan.setTraceId(rootSpan.getTraceId());
+        clientSpan.setId("2");
+        clientSpan.setParentId("root");
+        clientSpan.setName("post");
+
+        Span serverSpan = new Span(Arrays.asList(urlServiceABinAnn),
+                SpanServiceITest.serverAnnotations(
+                        SpanServiceITest.millisToMicroSeconds(7), SpanServiceITest.millisToMicroSeconds(10)));
+        serverSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(7));
+        serverSpan.setDuration(SpanServiceITest.calculateDuration(7L, 10L));
+        serverSpan.setTraceId(rootSpan.getTraceId());
+        serverSpan.setId("2");
+        serverSpan.setParentId("root");
+        serverSpan.setName("post");
+
+        post(Server.Zipkin, "/spans", null, Arrays.asList(rootSpan));
+        post(Server.Zipkin, "/spans", null, Arrays.asList(clientSpan));
+        post(Server.Zipkin, "/spans", null, Arrays.asList(serverSpan));
+
+        Wait.until(() -> {
+            Collection<CommunicationSummaryStatistics> communicationSummaryStatistics =
+                    analyticsService.getCommunicationSummaryStatistics(null, requestCriteria(), false);
+            return communicationSummaryStatistics == null || communicationSummaryStatistics.size() != 2 ? false : true;
+        }, 30, TimeUnit.SECONDS);
+
+        List<CommunicationSummaryStatistics> communicationSummaryStatisticsList = new ArrayList<>(
+                analyticsService.getCommunicationSummaryStatistics(null, requestCriteria(), false));
+        Assert.assertEquals(2, communicationSummaryStatisticsList.size());
+
+        CommunicationSummaryStatistics summaryStatistics = communicationSummaryStatisticsList.get(0);
+        Assert.assertEquals(EndpointUtil.encodeEndpoint("/root", "GET"), summaryStatistics.getId());
+        Assert.assertEquals(1, summaryStatistics.getOutbound().size());
+        Assert.assertTrue(summaryStatistics.getOutbound().containsKey(EndpointUtil.encodeEndpoint("/serviceA", "POST")));
+
```

Comment:
might also be worth checking the min/avg/max durations and count values. realise these are missing from the other tests :)